src/shared/components/Banners/Banner/Banner.tsx

```diff
@@ -0,0 +1,59 @@
+import { Alert } from '@mui/material';
+import { useEffect, useState } from 'react';
+import useWindowFocus from 'use-window-focus';
+
+import { Svg } from 'shared/components/Svg';
+import { StyledClearedButton } from 'shared/styles';
+
+import { BannerProps } from './Banner.types';
+
+export const Banner = ({
+  children,
+  duration = 5000,
+  onClose,
+  hasCloseButton = !!onClose,
+  severity,
+}: BannerProps) => {
+  let timeoutId: NodeJS.Timeout | undefined;
+  const [isHovering, setIsHovering] = useState(false);
+  const isWindowFocused = useWindowFocus();
+
+  // Close banner on timeout only while window is focused & banner not hovered
+  // (a11y behavior adapted from MUI SnackBar)
+  useEffect(() => {
+    if (!duration || !onClose || isHovering || !isWindowFocused) return;
+
+    timeoutId = setTimeout(onClose, duration);
+
+    return () => {
+      clearTimeout(timeoutId);
+    };
+  }, [duration, isHovering, isWindowFocused, onClose]);
+
+  return (
+    <Alert
+      iconMapping={{
+        info: getSvg('more-info-filled'),
+        success: getSvg('check-circle'),
+        warning: getSvg('exclamation-circle'),
+        error: getSvg('exclamation-octagon'),
+      }}
+      components={{
+        CloseButton: StyledClearedButton,
+      }}
+      componentsProps={{
+        closeButton: {
+          children: <Svg id="close" />,
+        },
+      }}
+      onClose={hasCloseButton ? onClose : undefined}
+      onMouseEnter={() => setIsHovering(true)}
+      onMouseLeave={() => setIsHovering(false)}
+      severity={severity}
+    >
+      {children}
+    </Alert>
+  );
+};
+
+const getSvg = (id: string) => <Svg id={id} width={32} height={32} />;
```

Comment:
I suppose no need to generate the same svg compononents each time a new banner is created. Could you please create the constants with respective components for them?

Reply:
Fixed!